test/test_helpers.py

```diff
@@ -432,6 +432,34 @@ def test_redq_make(device, from_pixels, gsde, exploration):
     del proof_environment
 
 
+def test_seed_generator():
+    num_tests = 5
+    num_seeds = 100
+    prev_seeds = []
+
+    def generate_seeds(seed, repeat):
+        seeds = []
+        for _ in range(repeat):
+            seed = seed_generator(seed)
+            seeds.append(seed)
+        return seeds
+
+    # Check unique seed generation
+    for initial_seed in range(num_tests):
+        seeds = generate_seeds(initial_seed, num_seeds)
+        assert len(seeds) == num_seeds
+        assert len(seeds) == len(set(seeds))
+        if prev_seeds:
+            assert set(prev_seeds) != set(seeds)
```

Comment:
great! but not super readable
I guess what we want is to test that the union is empty

what about
```
assert len(set(prev_seeds).union(seeds)) == 0
```